src/main/java/org/jenkinsci/remoting/protocol/IOHub.java

```diff
@@ -693,6 +694,13 @@ public void run() {
                         (ops & SelectionKey.OP_CONNECT) == SelectionKey.OP_CONNECT,
                         (ops & SelectionKey.OP_READ) == SelectionKey.OP_READ,
                         (ops & SelectionKey.OP_WRITE) == SelectionKey.OP_WRITE);
+            } catch (Throwable e) {
+                if (LOGGER.isLoggable(Level.SEVERE)) {
+                    LogRecord record = new LogRecord(Level.SEVERE, "[{0}] Listener {1} propagated an uncaught {2}");
+                    record.setThrown(e);
+                    record.setParameters(new Object[]{workerThread.getName(), listener, e.getClass().getSimpleName()});
+                    LOGGER.log(record);
+                }
```

Comment:
🐛 rethrow. Otherwise it suppresses <code>Error</code>s, which are supposed to be fatal here